cubedash/index/postgis/_api.py

```diff
@@ -376,39 +355,65 @@ def collections_search_query(
         bbox: tuple[float, float, float, float] | None = None,
         time: tuple[datetime, datetime] | None = None,
         q: Sequence[str] | None = None,
-    ) -> Result:
-        collection = self.collection_cols().subquery()
-        query = select(collection).where(collection.c.period_type == "all")
-
+    ) -> list[Row]:
+        # STAC Collections only hold a bounding box in EPSG:4326, no geometry
+        # Calculate the bounding box on the server, it's far more efficient
+        collection_bbox = func.Box2D(
+            func.ST_Transform(TimeOverview.footprint_geometry, 4326)
+        )
+        bbox_array = array(
+            [
+                func.ST_XMin(collection_bbox),
+                func.ST_YMin(collection_bbox),
+                func.ST_XMax(collection_bbox),
+                func.ST_YMax(collection_bbox),
+            ]
+        )
+        bbox_or_null = case(
+            (func.ST_XMin(collection_bbox).is_(None), None), else_=bbox_array
+        )
+        query = (
+            select(
+                ODC_PRODUCT.definition,
+                ProductSpatial.name,
+                bbox_or_null.label("bbox"),
+                ProductSpatial.time_earliest,
+                ProductSpatial.time_latest,
+            )
+            .select_from(TimeOverview)
+            .join(ProductSpatial, ProductSpatial.id == TimeOverview.product_ref)
+            .join(ODC_PRODUCT, ProductSpatial.name == ODC_PRODUCT.name)
+            .where(TimeOverview.period_type == "all")
+        )
         if name:
-            query = query.where(collection.c.name == name)
+            query = query.where(ProductSpatial.name == name)
 
         if bbox:
             query = query.where(
-                collection.c.footprint_geometry.intersects(func.ST_MakeEnvelope(*bbox))
+                TimeOverview.footprint_geometry.intersects(func.ST_MakeEnvelope(*bbox))
             )
 
         if time:
             query = query.where(
                 and_(
-                    default_utc(time[0]) <= default_utc(collection.c.time_latest),
-                    default_utc(collection.c.time_earliest) <= default_utc(time[1]),
+                    default_utc(time[0]) <= default_utc(ProductSpatial.time_latest),
+                    default_utc(ProductSpatial.time_earliest) <= default_utc(time[1]),
                 )
             )
 
         if q:
             title = SimpleDocField(
                 name="title",
                 description="product title",
-                alchemy_column=collection.c.definition,
+                alchemy_column=ODC_PRODUCT.definition,
                 indexed=False,
                 offset=("metadata", "title"),
             )
 
             description = SimpleDocField(
                 name="description",
                 description="product description",
-                alchemy_column=collection.c.definition,
+                alchemy_column=ODC_PRODUCT.definition,
```

Comment:
The corresponding postgres change also fixes the type of the `offset=` parameter.

Reply:
I think that `SimpleDocField` turns a plain string into a length 1 tuple... but I'm not sure.

This is ugly though, I'll make them match.

Reply:
Thanks!

pyright was complaining, but mypy wasn't for some reason.

It's fixed now regardless.